### PR TITLE
Check for backend support in Jax context parallel fused attention test

### PR DESCRIPTION
**Pull request by mgoldfarb-nvidia**
# Description

Adds additional checks to avoid running unsupported datatypes or shardings of fused attention on Jax.

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refractor

## Changes

Skip tests unless on SM 90+

# Checklist:

- [X] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [X] The functionality is complete
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: transformer_engine/common/fused_attn/fused_attn.cpp [hunk 1]
```diff
@@ -136,6 +136,8 @@ NVTE_Fused_Attn_Backend nvte_get_fused_attn_backend(
          // d=256 only supported for forward
          (sm_arch_ >= 90 && cudnn_runtime_version >= 90000 && head_dim_qk <= 256 &&
           head_dim_qk % 8 == 0 && head_dim_v <= 256 && head_dim_v % 8 == 0)) &&
+        // bfloat support in SM80+
+        ((q_dtype == NVTEDType::kNVTEFloat16 || sm_arch_ >= 80)) &&
```
[the remark this answers is not in the document]
Comment (mgoldfarb-nvidia) [reply transformer_engine/common/fused_attn/fused_attn.cpp]: Removed.

File: transformer_engine/jax/cpp_extensions/attention.py [hunk 2]
```diff
@@ -923,26 +923,30 @@ def check_supported(self):
         header = "Context parallel fused attention"
 
         allowed_layouts = [NVTE_QKV_Layout.NVTE_BSHD_BS2HD, NVTE_QKV_Layout.NVTE_BSHD_BSHD_BSHD]
-        assert self.config.qkv_layout in allowed_layouts, (
-            f"{header} only supports layouts: {','.join([str(x) for x in allowed_layouts])} got:"
-            f" {self.config.qkv_layout}"
-        )
+        if self.config.qkv_layout not in allowed_layouts:
```
Comment (phu0ngng) [inline transformer_engine/jax/cpp_extensions/attention.py:926]: Hi, why do we replace `assert` with `if not` here?

File: transformer_engine/jax/cpp_extensions/misc.py [hunk 3]
```diff
@@ -167,3 +170,8 @@ def is_ffi_enabled():
     is_enabled = int(os.getenv("NVTE_JAX_WITH_FFI", "1"))
     assert is_enabled in (0, 1), "Invalid NVTE_JAX_WITH_FFI value"
     return is_supported and is_enabled
+
+
+def is_bf16_compatible() -> None:
```
Comment (phu0ngng) [inline transformer_engine/jax/cpp_extensions/misc.py:175]: I have not yet found where this helper is used.

File: transformer_engine/jax/cpp_extensions/attention.py (hunk 2 above)
Comment (mgoldfarb-nvidia) [reply transformer_engine/jax/cpp_extensions/attention.py:926]: In a previous review it was suggested that asserts could be disabled and we should instead raise a proper exception.

File: transformer_engine/jax/cpp_extensions/misc.py (hunk 3 above)
Comment (mgoldfarb-nvidia) [reply transformer_engine/jax/cpp_extensions/misc.py]: Removed - stale code from earlier

**Review by zlsh80826 — APPROVED** [review]
Local tested that `tests/jax/test_distributed_fused_attn.py` passed on both H100 and V100 on 10-02 image.
LGTM!

**Pull request merged**